src/border_agent/border_agent.cpp

```diff
@@ -163,6 +165,9 @@ BorderAgent::BorderAgent(otbr::Ncp::RcpHost &aHost, Mdns::Publisher &aPublisher)
     , mBaseServiceInstanceName(OTBR_MESHCOP_SERVICE_INSTANCE_NAME)
 {
     mHost.AddThreadStateChangedCallback([this](otChangedFlags aFlags) { HandleThreadStateChanged(aFlags); });
+    mIsEphemeralKeyEnabled = (otThreadGetVersion() >= kThreadVersion14);
```

Comment:
Another review concern is too many macro; I suppose the target of macro is easy to config, so in the new commit, add a kEphemeralKeyEnableOnInit for config; Let me know your concerns. Thanks!

Reply:
> Another review concern is too many macro

IMHO the suggested macro is not that harmful because it doesn't guard code blocks. Instead, it's just used as a configurable boolean value.

Reply:
Have modified to Macro accordingly; Thanks for re-reviewing